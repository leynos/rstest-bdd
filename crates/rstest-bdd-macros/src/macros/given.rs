//! Implementation of the `#[given]` macro.

use proc_macro::TokenStream;

/// Macro for defining a Given step that registers with the step inventory.
pub(crate) fn given(attr: TokenStream, item: TokenStream) -> TokenStream {
    super::step_attr(attr, item, crate::StepKeyword::Given)
}
