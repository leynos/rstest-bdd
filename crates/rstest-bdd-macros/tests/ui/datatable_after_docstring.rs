//! Compile-fail fixture: `#[datatable]` must precede a doc string parameter.

use rstest_bdd_macros::given;

#[given("datatable after docstring")]
fn step(docstring: String, #[datatable] table: Vec<Vec<String>>) {}

fn main() {}
