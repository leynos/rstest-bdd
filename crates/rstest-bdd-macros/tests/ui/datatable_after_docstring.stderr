error: invalid step function signature: DataTable must be declared before DocString to match Gherkin ordering
 --> tests/ui/datatable_after_docstring.rs:6:41
  |
6 | fn step(docstring: String, #[datatable] table: Vec<Vec<String>>) {}
  |                                         ^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: Use `#[given] fn name(ctx: &rstest_bdd::StepContext, ...)` and valid fixtures.
